pallets/moonbeam-orbiters/src/lib.rs

```diff
@@ -0,0 +1,485 @@
+		type AddCollatorOrigin: EnsureOrigin<Self::Origin>;
+
+		/// The currency type.
+		type Currency: NamedReservableCurrency<Self::AccountId>;
+
+		/// Origin that is allowed to remove a collator from orbiters program.
+		type DelCollatorOrigin: EnsureOrigin<Self::Origin>;
+
+		/// Maximum number of orbiters per collator.
+		type MaxPoolSize: Get<u32>;
+
+		/// Maximum number of round to keep on storage.
+		type MaxRoundArchive: Get<Self::RoundIndex>;
+
+		/// Reserve identifier for this pallet instance.
+		type OrbiterReserveIdentifier: Get<ReserveIdentifierOf<Self>>;
+
+		/// Number of rounds before changing the selected orbiter.
+		type RotatePeriod: Get<Self::RoundIndex>;
+
+		/// Round index type.
+		type RoundIndex: Parameter
+			+ Member
+			+ MaybeSerializeDeserialize
+			+ sp_std::fmt::Debug
+			+ Default
+			+ sp_runtime::traits::MaybeDisplay
+			+ sp_runtime::traits::AtLeast32Bit
+			+ Copy;
+
+		/// Origin that is allowed to update the minimal orbiter deposit amount.
+		type UpdateMinOrbiterDepositOrigin: EnsureOrigin<Self::Origin>;
+	}
+
+	#[pallet::storage]
+	#[pallet::getter(fn account_lookup_override)]
+	/// Account lookup override
+	pub type AccountLookupOverride<T: Config> =
+		StorageMap<_, Blake2_128Concat, T::AccountId, Option<T::AccountId>>;
+
+	#[pallet::storage]
+	#[pallet::getter(fn collators_pool)]
+	/// Current orbiters, with their "parent" collator
+	pub type CollatorsPool<T: Config> =
+		CountedStorageMap<_, Blake2_128Concat, T::AccountId, CollatorPoolInfo<T::AccountId>>;
+
+	#[pallet::storage]
+	/// Current round index
+	pub(crate) type CurrentRound<T: Config> = StorageValue<_, T::RoundIndex, ValueQuery>;
+
+	#[pallet::storage]
+	#[pallet::getter(fn min_orbiter_deposit)]
+	/// Minimum deposit required to be registered as an orbiter
+	pub type MinOrbiterDeposit<T: Config> = StorageValue<_, BalanceOf<T>, ValueQuery>;
+
+	#[pallet::storage]
+	/// Store active orbiter per round and per parent collator
+	pub(crate) type OrbiterPerRound<T: Config> = StorageDoubleMap<
+		_,
+		Twox64Concat,
+		T::RoundIndex,
+		Blake2_128Concat,
+		T::AccountId,
+		T::AccountId,
+		OptionQuery,
+	>;
+
+	#[pallet::genesis_config]
+	pub struct GenesisConfig<T: Config> {
+		pub min_orbiter_deposit: BalanceOf<T>,
+	}
+
+	#[cfg(feature = "std")]
+	impl<T: Config> Default for GenesisConfig<T> {
+		fn default() -> Self {
+			Self {
+				min_orbiter_deposit: One::one(),
+			}
+		}
+	}
+
+	#[pallet::genesis_build]
+	impl<T: Config> GenesisBuild<T> for GenesisConfig<T> {
+		fn build(&self) {
+			assert!(
+				self.min_orbiter_deposit > Zero::zero(),
+				"Minumal orbiter deposit should be greater tham zero"
+			);
+			MinOrbiterDeposit::<T>::put(self.min_orbiter_deposit)
+		}
+	}
+
+	#[pallet::hooks]
+	impl<T: Config> Hooks<BlockNumberFor<T>> for Pallet<T> {
+		fn on_initialize(_: T::BlockNumber) -> Weight {
+			// Prune old OrbiterPerRound entries
+			let current_round = CurrentRound::<T>::get();
```

Comment:
This requires migration of the current round value

Reply:
In reality it doesn't depend on the staking, it depends on the notion of round injected by the runtime, defined by the callback `on_new_round()`.
It would work the same way with a consensus other than PoS.

Reply:
I think we don't need migration, because now the round index is retrieved from the runtime